library/Icinga/Application/ProvidedHook/DbMigration.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/* Icinga Web 2 | (c) 2023 Icinga GmbH | GPLv2+ */
+
+namespace Icinga\Application\ProvidedHook;
+
+use Icinga\Application\Hook\MigrationHook;
+use Icinga\Common\Database;
+use Icinga\Model\Schema;
+use ipl\Orm\Query;
+
+class DbMigration extends MigrationHook
+{
+    use Database;
+
+    public function getName(): string
+    {
+        return $this->translate('Icinga Web');
+    }
+
+    public function providedDescriptions(): array
+    {
+        return [];
+    }
+
+    public function getVersion(): string
+    {
+        if ($this->version === null) {
+            $conn = $this->getDb();
+            $schemaQuery = $this->getSchemaQuery()
+                ->orderBy('id', SORT_DESC)
+                ->limit(2);
+
+            if (static::getColumnType($conn, $schemaQuery->getModel()->getTableName(), 'success')) {
+                /** @var Schema $schema */
+                foreach ($schemaQuery as $schema) {
+                    if ($schema->success) {
+                        $this->version = $schema->version;
+
+                        break;
+                    }
+                }
+
+                if (! $this->version) {
+                    $this->version = '2.12.0';
+                }
+            } elseif (static::tableExists($conn, $schemaQuery->getModel()->getTableName())) {
```

Comment:
The table check isn't reliable, since the setup wizard didn't define the correct permissions for it till now. I'd suggest to check the collation of e.g. the `icingaweb_user_preference.username` column.

Reply:
Collation is MYSQL specific.

Reply:
Not a problem. Then perform a different check depending on the database vendor. The collation check is only required for MySQL anyway, since there the issue with the grants exist. For postgres this is probably not the case, so there you can check for the schema tables existence.